File: codingChallenges.js
const readline = require("readline");

function challengeOne() {
	const rl = readline.createInterface({
		input: process.stdin,
		output: process.stdout,
	});

	rl.question("Please enter a two numbers (separated by a space): ", (ans) => 

		{
			let inputs = ans.split(" ");
			if(inputs.length == 2){
				let sum = addition(inputs[0], inputs[1]);
				if(isNaN(sum)){
					console.log('Invalid arguments. Must be two numbers.');
				} else {		
					console.log(`${inputs[0]} + ${inputs[1]} = ${sum}`)
				}
			} else {
				console.log('Invalid arguments. Must be two numbers.');
			}
			rl.close();
		});
}
function addition(a, b) {
	let aNum = parseFloat(a);
	let bNum = parseFloat(b);
	return aNum + bNum;	
}

function challengeTwo() {
	console.log("Increment 0: " + addition(0)); // 1
	let a = 54;
	console.log("a start: " + a); // 54
	a = addition(a);
	a = addition(a);
	console.log("a end: " + a); // 56
}

function addition(a) {
	return a + 1;
}


// challengeOne();
challengeTwo();

File: CodingChallenges.py
def addition(a, b):
	return a + b

def tri_area(a: int, b: int) -> int:
	if( a <= 0 or b <= 0):
		raise Exception("Arguments must be postive values greater than 0.")
	else:
		return round((a * b) / 2)

def challengeOne():
	try:
		a = int(input("Enter the first number: "))
		b = int(input("Enter the second number: "))
		print(f'{a} + {b} = {addition(a,b)}')
	except ValueError:
		print('Invalid arguments. Only integers are accepted.')


def challengeTwo():
	try:
		a = int(input("Enter the base: "))
		b = int(input("Enter the height: "))
		print(f'area = {tri_area(a,b)}')
	except ValueError:
		print('Invalid arguments. Only integers are accepted.')
	
if __name__ == "__main__":
	challengeOne()
	challengeTwo()
	